packages/inspector-extension/src/index.ts

```diff
@@ -89,11 +89,11 @@ const inspector: JupyterLabPlugin<IInspector> = {
       execute: () => {
         if (!manager.inspector || manager.inspector.isDisposed) {
           manager.inspector = newInspectorPanel();
-          shell.addToMainArea(manager.inspector);
         }
-        if (manager.inspector.isAttached) {
-          shell.activateById(manager.inspector.id);
+        if (!manager.inspector.isAttached) {
+          shell.addToMainArea(manager.inspector.parent, { activate: false });
```

Comment:
Going through the parent doesn't seem particularly clean here. Why not have `newInspectorPanel` return the `MainAreaWidget` that is created, and operate with that in the `execute()` function?

Reply:
Sure - would you be able to take that up in a new PR? I didn't concentrate a lot on this bit of the PR.

Reply:
Sure.

Reply:
@ian-r-rose - did you make an issue for your comment?

Reply:
If memory serves, I made a branch to fix this, and found that it would be rather more work to refactor it than I had planned, so I dropped it.